cmake/install_helpers.cmake

```diff
@@ -68,10 +68,11 @@ function(ginkgo_install)
 
     # install the public header files
     install(DIRECTORY "${Ginkgo_SOURCE_DIR}/include/"
-        DESTINATION "${CMAKE_INSTALL_FULL_INCLUDEDIR}"
-        COMPONENT Ginkgo_Development
-        FILES_MATCHING PATTERN "*.hpp"
-        )
+            DESTINATION "${CMAKE_INSTALL_FULL_INCLUDEDIR}"
+            COMPONENT Ginkgo_Development
+            FILES_MATCHING PATTERN "*.hpp"
+            PATTERN "extensions" EXCLUDE
```

Comment:
I thought this would go into a new `COMPONENT` ? Like `Ginkgo_Extensions` ?

Reply:
TBH I still don't quite understand the cmake components, so I didn't touch that. But that can be revisited, when we decided how to handle extensions in general.